File: DATA_STRUCTURE/MORNINGSESSION/ARRAY/rotateArray.java
class Solution{
	public void rotateArray(int arr[],int n,int k){
		for(int i=0 ; i<k ; i++){
			int last = arr[n];
			for(int j=n ; j>0 ; j--){
				arr[j] = arr[j-1];
			}
			arr[0] = last;
		}
	}
	public static void main(String[]args){
		Solution s = new Solution();
		int arr[] = {1,2,3,4,5,6,7};
		int n = 7;
		int k = 3;
		System.out.println("Array Before Rotate : ");
		for(int i=0 ; i<n ; i++){
			System.out.print("|"+arr[i]+"|");
		}
		s.rotateArray(arr,n-1,k);
		System.out.println("\nArray After Rotate : ");
		for(int i=0 ; i<n ; i++){
			System.out.print("|"+arr[i]+"|");
		}
		System.out.println();
	}
}

File: DATA_STRUCTURE/ASSIGNMENT/RECURSION/rotateArray.c
#include<stdio.h>

int rotateArray(int arr[],int n,int B,int i,int j,int last){

	if(i >= B ) return 0;

	if(j < 0){
		j = n;
		i = i+1;
		arr[0] = last;
		last = arr[n];
	}

	if(i<B)
		arr[j] = arr[j-1];

	return rotateArray(arr,n,B,i,j-1,last);
}

void main(){
	int arr[] = {1,2,3,4};
	int n = 4;
	int B = 2;
	int i = 0;
	int j = n-1;
	int last = arr[j];
	rotateArray(arr,n-1,B,i,j,last);
	for(int i=0 ; i<n ; i++){
		printf(" %d ",arr[i]);
	}
	printf("\n");
}
